web_external/js/views/map/MapAdapters.js

```diff
@@ -0,0 +1,181 @@
+/** */
+minerva.models.MapLayerModel = Backbone.Model.extend({
+    /** */
+    initialize: function (settings) {
+        this.dataset = settings.dataset;
+        this.adapter = settings.adapter;
+        // => MapLayerView
+        this.geoJsLayer = settings.geoJsLayer;
+        var opacity = _.isUndefined(settings.opacity) || _.isNull(settings.opacity) ? 1 : settings.opacity;
+        // => MapLayerView
+        this.geoJsLayer.opacity(opacity);
+    },
+
+    /** */
+    renderable: function () {
+        this.trigger('m:map_layer_renderError');
+    },
+
+    /** */
+    deleteLayer: function (geoJsMap) {
+        // => MapLayerView
+        geoJsMap.deleteLayer(this.geoJsLayer);
+    },
+
+    /** */
+    setOpacity: function (opacity) {
+        this.geoJsLayer.opacity(opacity);
+    }
+});
+
+/** */
+minerva.models.ReaderMapLayerModel = minerva.models.MapLayerModel.extend({
+    defaults: {
+        readerType: 'jsonReader'
+    },
+
+    uponGeoDataLoaded: function () {
+        try {
+            var reader = geo.createFileReader(this.get('readerType'), {layer: this.geoJsLayer});
+            reader.read(this.dataset.get('geoData'), _.bind(function () {
+                this.trigger('m:map_layer_renderable');
+            }, this));
+        } catch (err) {
+            console.error('This dataset cannot be rendered to the map');
+            console.error(err);
+            this.dataset.set('geoError', true);
+            // => MapLayerView
+            this.geoJsLayer.clear();
+            this.trigger('m:map_layer_renderError');
+        }
+    },
+
+    /** */
+    renderable: function () {
+        this.dataset.once('m:dataset_geo_dataLoaded', function () {
+            this.uponGeoDataLoaded();
+        }, this);
+        this.dataset.loadGeoData();
+    }
+});
+
+/** */
+minerva.models.GeojsonMapLayerModel = minerva.models.ReaderMapLayerModel.extend({});
```

Comment:
I'm a little confused here; isn't it the role of the adapter to obfuscate the underlying data type?